roles/openshift_health_checker/openshift_checks/__init__.py

```diff
@@ -19,15 +19,23 @@ class OpenShiftCheckException(Exception):
 
 @six.add_metaclass(ABCMeta)
 class OpenShiftCheck(object):
-    """A base class for defining checks for an OpenShift cluster environment."""
+    """
+    A base class for defining checks for an OpenShift cluster environment.
+
+    Expect a method execute_module, optional dict task_vars, and optional string tmp.
+    execute_module is expected to have a signature compatible with _execute_module
+    from ansible plugins/action/__init__.py, e.g.:
+    def execute_module(module_name=None, module_args=None, tmp=None, task_vars=None, **kwargs)
+    This is stored so that it can be invoked in subclasses as check.execute_module("name", args)
+    which provides the check's task_vars and tmp by default.
+    """
 
-    def __init__(self, execute_module=None, module_executor=None):
-        if execute_module is module_executor is None:
-            raise TypeError(
-                "__init__() takes either execute_module (recommended) "
-                "or module_executor (deprecated), none given")
-        self.execute_module = execute_module or module_executor
-        self.module_executor = self.execute_module
+    def __init__(self, execute_module, task_vars=None, tmp=None):
```

Comment:
It is possible to do the refactor gradually without breaking on-going PRs, but if you want to avoid that "exercise" I'm fine with it.

Perhaps remove `module_executor` in a separate commit?